### ISSUE TITLE
Support for changing the form returned in ConvertKit::get_category_form()

### ISSUE THREAD

**Issue by danielbachhuber** [ConvertKit/convertkit-wordpress#263]
We're using the "This form will be automatically added to posts in this category." feature:

<img width="878" alt="image" src="https://user-images.githubusercontent.com/36432/141022026-6233cda0-dc2e-4e37-8758-ba0ada5c425c.png">

When a post has an Instant Pot category _and_ a Vegetarian category, we'd like the Instant Pot form to take precedence. However, the Vegetarian form does because the Vegetarian category was created earlier.

It would be great if `::get_category_form()` had a filter so I could modify its behavior for this edge case:

```php
/**
 * Get default form for the post's categories.
 * If there are multiple set the last one found will be returned.
 *
 * @param int $id
 * @return int $form_id
 */
private static function get_category_form( $id ) {
	$categories = wp_get_post_categories( $id );

	$form_id = 0;
	foreach ( $categories as $category ) {

		$id = get_term_meta( $category, 'ck_default_form', true );
		if ( $id ) {
			$form_id = $id;
		}
	}

	return apply_filters( 'convertkit_get_category_form', $form_id, $id );
}
```

At the moment, I have to do some hacky filtering:

```php
add_filter(
	'the_content',
	function( $content ) {
		if ( is_single() ) {
			// Prioritize the Instant Pot niche sequence vs. Vegetarian.
			if ( false !== stripos( $content, 'e4ec822f1c' )
				&& has_term( 'vegetarian', 'category' )
				&& has_term( 'instant-pot', 'category' ) ) {
				$bits          = explode( PHP_EOL, $content );
				$last          = count( $bits ) - 1;
				$bits[ $last ] = str_replace(
					'e4ec822f1c', // Vegetarian list.
					'bb9b33958d', // Instant Pot list.
					$bits[ $last ]
				);
				$content       = implode( PHP_EOL, $bits );
			}
		}
		return $content;
	},
	11
);
```

Thanks for the consideration!

**Comment by n7studios** [comment]
Great suggestion.  This was already on my radar to include in the next update, so good to know it's needed.

Filter name will be a little different than proposed - `convertkit_output_append_form_to_content_form_id`, due to some underlying structural changes and naming conventions - but parameters will be the same (`$form_id`, `$post_id`).

I'll leave this open for now until the update adds this in.

**Comment by danielbachhuber** [comment]
👍 Sounds good, thanks!

**Comment by n7studios** [comment]
Update: this is included in 1.9.6, due for release in December:
https://github.com/ConvertKit/convertkit-wordpress/blob/1.9.6/ACTIONS-FILTERS.md#convertkit_output_append_form_to_content_form_id

**Comment by n7studios** [comment]
1.9.6 now released at https://en-gb.wordpress.org/plugins/convertkit

Specific filter can be found at https://github.com/ConvertKit/convertkit-wordpress/blob/master/ACTIONS-FILTERS.md#convertkit_output_append_form_to_content_form_id
